python/modules/IcePy/Operation.cpp

```diff
@@ -703,19 +703,22 @@ Operation::marshalResult(Ice::OutputStream& os, PyObject* result)
         PyObject* arg = PyTuple_GET_ITEM(resultTuple.get(), info->pos);
         if ((!info->optional || arg != Py_None) && !info->type->validate(arg))
         {
-            try
-            {
-                throwPythonException();
-            }
-            catch (const Ice::UnknownException& ex)
+            // TODO: Provide the parameter name instead?
+            ostringstream ostr;
+            ostr << "cannot marshal result: invalid value for out argument " << (info->pos + 1) << " in operation '"
```

Comment:
Can we improve the error message? Invalid is a bit ambiguous to me. Doe it mean wrong/unexpected type, a value that's out of range, something else?